services/user.py

```
@@ -0,0 +1,52 @@
+from typing import Any
+
+from django.contrib.auth import get_user_model
+from django.db import transaction
+
+
+@transaction.atomic
+def create_user(
+    username: str,
+    password: str,
+    email: str = None,
+    first_name: str = None,
+    last_name: str = None,
+) -> Any:
+    user = get_user_model().objects.create_user(
+        username=username, password=password
+    )
+    if email is not None:
+        user.email = email
+    if first_name is not None:
+        user.first_name = first_name
+    if last_name is not None:
+        user.last_name = last_name
+    user.save()
+    return user
+
+
+def get_user(user_id: int) -> Any:
+    return get_user_model().objects.get(pk=user_id)
+
+
+def update_user(
+    user_id: int,
+    username: str = None,
+    password: str = None,
+    first_name: str = None,
+    last_name: str = None,
+    email: str = None,
+) -> None:
+    if (user := get_user(user_id)) is None:
```

`get_user` uses `get(pk=user_id)`, which raises an exception if the user does not exist. The check `if (user := get_user(user_id)) is None:` will never be True and thus is ineffective. Either remove the None check and let the exception propagate (if that's the desired behavior), or handle the missing user explicitly by catching `get_user_model().DoesNotExist` and returning/raising a clearer error. This will avoid unexpected exceptions from bubbling up when `update_user` is called with a non-existent id.